src/sinks/util/request_builder.rs

```diff
@@ -1,10 +1,18 @@
-use std::io;
+use std::{io, num::NonZeroUsize};
 
 use bytes::Bytes;
 use vector_common::request_metadata::{GroupedCountByteSize, RequestMetadata};
 
 use super::{encoding::Encoder, metadata::RequestMetadataBuilder, Compression, Compressor};
 
+/// Default concurrency limit for a request builder
+const DEFAULT_REQUEST_BUILDER_CONCURRENCY_LIMIT: Option<NonZeroUsize> = NonZeroUsize::new(64);
```

Comment:
Related to @lukesteensen's comment, could we make this an undocumented environment variable? It might be useful to have this knob when troubleshooting customer issues. We could add `_EXPERIMENTAL_` in its name to make it extra clear it isn't meant to be depended on.

Reply:
That's a great idea. I'll add it.